tests/test_schema.py

```diff
@@ -3001,3 +3002,27 @@ class Model(BaseModel):
             },
         },
     }
+
+
+def test_sub_model_with_type_attributes():
+    class Foo(BaseModel):
+        b: float
+
+    class Bar(BaseModel):
+        a: int
+        b: Type[Foo]
+
+    assert Bar.schema() == {
+        'title': 'Bar',
+        'type': 'object',
+        'properties': {'a': {'title': 'A', 'type': 'integer'}, 'b': {'$ref': '#/definitions/Foo'}},
+        'required': ['a', 'b'],
+        'definitions': {
+            'Foo': {
+                'title': 'Foo',
+                'type': 'object',
+                'properties': {'b': {'title': 'B', 'type': 'number'}},
+                'required': ['b'],
+            }
+        },
+    }
```

Comment:
I've tested this a bit manually, not sure if there are any more tests to be added for it.